src/plone/restapi/services/workflow/info.py

```diff
@@ -28,6 +29,13 @@ def __call__(self, expand=False):
         if not expand:
             return result
 
+        # Prevent 404 on site root on workflow request
+        # Although 404 will be more semantic, for the sake of uniformity of the
+        # API we fake the response to the endpoint by providing an empty
+        # response instead of a 404.
+        if IPloneSiteRoot.providedBy(self.context):
+            result['workflow'].update({'history': [], 'transitions': []})
```

Comment:
Immediately `return result` here? No point in even trying all the rest, right? That would make it clearer what happens here.

Reply:
Fixed. Thx for the hint!